core/src/jsMain/kotlin/Peripheral.kt

```diff
@@ -310,8 +310,8 @@ public class JsPeripheral internal constructor(
     private var isDisconnectedListenerRegistered = false
     private val disconnectedListener: (JsEvent) -> Unit = {
         logger.debug { message = GATT_SERVER_DISCONNECTED }
-        connectJob.cancel()
-        disconnectJob.launch()
+        @Suppress("DeferredResultUnused") // Safe to ignore Deferred because the result is shared elsewhere
+        disconnectJob.getOrAsync()
```

Comment:
And here, strongly suggest the `connectJob.cancel()` remain here and move that change to a separate PR.